csi/controller_server.go

```diff
@@ -678,17 +769,32 @@ func (cs *ControllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
 		return nil, status.Error(codes.InvalidArgument, "missing snapshot id in request")
 	}
 
-	_, backupVolumeName, backupName := decodeSnapshotID(snapshotID)
-	backupVolume, err := cs.apiClient.BackupVolume.ById(backupVolumeName)
-	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
-	}
-
-	if backupVolume != nil && backupVolume.Name != "" {
-		backupVolume, err = cs.apiClient.BackupVolume.ActionBackupDelete(backupVolume, &longhornclient.BackupInput{Name: backupName})
+	csiSnapshotType, sourceVolumeName, id := decodeSnapshotID(snapshotID)
+	if csiSnapshotType == csiSnapshotTypeLonghornSnapshot {
+		volume, err := cs.apiClient.Volume.ById(sourceVolumeName)
 		if err != nil {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
+		if volume != nil {
```

Comment:
THe volume should be `attached`.

Reply:
If the volume is not attached. API will return error